examples/flutter_gallery/lib/welcome/home.dart

```diff
@@ -0,0 +1,169 @@
+                    itemCount: _kSteps,
+                    itemBuilder: (BuildContext context, int index) => _steps[index].widget,
+                    controller: _pageController,
+                    onPageChanged: (int page) {
+                      _tabController.animateTo(page);
+                      _steps[page].key.currentState.animate(restart: true);
+                    },
+                  ),
+                ),
+              ),
+              Align(
+                alignment: FractionalOffset.bottomCenter,
+                child: Padding(
+                  padding: EdgeInsets.only(bottom: footerHeight + 20.0),
+                  child: _pageIndicator(),
+                ),
+              ),
+              Positioned.fill(
+                top: null,
+                child: Column(
+                  mainAxisSize: MainAxisSize.min,
+                  crossAxisAlignment: CrossAxisAlignment.stretch,
+                  children: <Widget>[
+                    Container(
+                      height: footerButtonHeight,
+                      child: FlatButton(
+                        shape: const BeveledRectangleBorder(
+                          borderRadius: BorderRadius.only(
+                            topLeft: Radius.circular(16.0),
+                            topRight: Radius.circular(16.0),
+                          ),
+                        ),
+                        color: footerColor,
+                        child: const Text(
+                          'START EXPLORING',
+                          style: footerButtonTextStyle,
+                        ),
+                        onPressed: () {
+                          _autoProgressTimer.cancel();
+                          if (widget.onDismissed != null) {
+                            widget.onDismissed();
+                          }
+                        },
+                      ),
+                    ),
+                    Container(
+                      color: footerColor,
+                      height: mediaQueryData.padding.bottom,
+                    )
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  TabPageSelector _pageIndicator() {
+    return TabPageSelector(
+      controller: _tabController,
+      color: const Color(0x99ffffff),
+      selectedColor: const Color(0xffffffff),
+    );
+  }
+
+  Timer _scheduleAutoProgressStepTimer() {
+    return Timer.periodic(Duration(seconds: _kAutoProgressSeconds), (_) {
+      int nextPage = _pageController.page.ceil() + 1;
+      if (nextPage == _kSteps) {
+        nextPage = 0;
+      }
+      _pageController.animateToPage(
+        nextPage,
+        duration: Duration(milliseconds: _kAutoProgressTransitionMilliseconds),
+        curve: Curves.easeInOut,
+      );
+    });
+  }
+}
+
+class _WelcomeStep<T extends StatefulWidget> {
+  _WelcomeStep(this.key, this.widget);
+  final GlobalKey<WelcomeStepState<T>> key;
```

Comment:
You can even further separate concerns by just having this do stuff instead of giving the key back.